core/server/server.spec.js

```diff
@@ -3,162 +3,336 @@
+
+      it('should require private influx config when influx configuration is enabled', function() {
+        delete customConfig.private.metrics.influx
+        expect(
+          () => new Server(customConfig, requiredInstanceMetadata)
+        ).to.throw(
+          'Private configuration is invalid. Check these paths: metrics,influx'
+        )
+      })
+
+      it('should require private metrics config when influx configuration is enabled', function() {
+        delete customConfig.private.metrics
+        expect(
+          () => new Server(customConfig, requiredInstanceMetadata)
+        ).to.throw(
+          'Private configuration is invalid. Check these paths: metrics'
+        )
+      })
+
+      it('should require username when influx configuration is enabled', function() {
+        delete customConfig.private.metrics.influx.username
+        expect(
+          () => new Server(customConfig, requiredInstanceMetadata)
+        ).to.throw(
+          'Private configuration is invalid. Check these paths: metrics,influx,username'
+        )
+      })
+
+      it('should require password when influx configuration is enabled', function() {
+        delete customConfig.private.metrics.influx.password
+        expect(
+          () => new Server(customConfig, requiredInstanceMetadata)
+        ).to.throw(
+          'Private configuration is invalid. Check these paths: metrics,influx,password'
+        )
+      })
+
+      it('should allow other private keys', function() {
+        customConfig.private.gh_token = 'my-token'
+        expect(
+          () => new Server(customConfig, requiredInstanceMetadata)
+        ).to.not.throw()
+      })
     })
+    describe('instance metadata', function() {
+      let customInstanceMetadata
 
-    expect(statusCode).to.equal(302)
-    // This value is set in `config/test.yml`
-    expect(headers.location).to.equal('http://frontend.example.test')
-  })
+      beforeEach(function() {
+        customInstanceMetadata = {
+          id: 'instance 3',
+          env: 'shields-io',
+          hostname: 's3.shields.io',
+        }
+      })
+
+      it('should not require id', function() {
+        delete customInstanceMetadata.id
+        expect(
+          () => new Server(config.util.toObject(), customInstanceMetadata)
+        ).to.not.throw()
+      })
+
+      it('should require hostname', function() {
+        delete customInstanceMetadata.hostname
+        expect(
+          () => new Server(config.util.toObject(), customInstanceMetadata)
+        ).to.throw('"hostname" is required')
+      })
 
-  it('should return the 410 badge for obsolete formats', async function() {
-    const { statusCode, body } = await got(`${baseUrl}npm/v/express.jpg`, {
-      throwHttpErrors: false,
+      it('should require env', function() {
+        delete customInstanceMetadata.env
+        expect(
+          () => new Server(config.util.toObject(), customInstanceMetadata)
+        ).to.throw('"env" is required')
```

Comment:
I can imagine these config and metadata tests providing a lot of confidence during development. Although, I'm not sure they seem worth keeping / maintaining. I'm open to being convinced.

Or, maybe the issue is that we should keep the tests, but the validation logic is in the wrong place, and that where it really belongs is in the influx object.

Reply:
I would like to keep them, because they are quite simple (but test more complicated cases using `.when()`) and shouldn't require a lot of work in the future. If you prefer I can try to parametrise them - this should shorten them. 

We can also move these tests to a separate file, but I'm not sure influx object is a good place for them. A config validation is performed in the sever object and tests for validation can be located in some file connected with the server.

Reply:
Let's leave it as is and consider it in a follow-on. If the validation really belongs in the server and not in the metrics, maybe it's time to break out the config into a separate module.

Reply:
I will add this as a TODO task in the end of the PR description.